src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -555,75 +529,109 @@ public SearchResult executeFQNPropertyTypeSearch( @RequestBody FQNSearchTerm sea
+                                                .filter( entry -> linkedEntitySetIds.contains( entry.getKey() ) )
+                                                .collect( Collectors.toMap( Map.Entry::getKey, Map.Entry::getValue ) ),
+                                        filter.getSrcEntitySetIds(),
+                                        filter.getDstEntitySetIds(),
+                                        filter.getAssociationEntitySetIds()
+                                ),
+                                principals ) );
             }
         }
 
 
         /* audit */
 
-        SetMultimap<UUID, UUID> neighborsByEntitySet = HashMultimap.create();
+        Map<UUID, Set<UUID>> neighborsByEntitySet = Maps.newHashMap();
 
         result.values().forEach( associationMap ->
-                associationMap.forEach( (associationEsId, association) -> {
-                    association.forEach( (neighborEsId, neighbor) -> {
-                        neighborsByEntitySet
-                                .put( associationEsId, neighbor.getAssociationEntityKeyId() );
-                        neighborsByEntitySet
-                                .put( neighborEsId, neighbor.getNeighborEntityKeyId() );
+                associationMap.forEach( ( associationEsId, association ) -> {
+                    if ( !neighborsByEntitySet.containsKey( associationEsId ) ) {
+                        neighborsByEntitySet.put( associationEsId, Sets.newHashSet() );
+                    }
+                    association.forEach( ( neighborEsId, neighbors ) -> {
+                        neighborsByEntitySet.putIfAbsent( neighborEsId, Sets.newHashSet() );
+                        neighbors.forEach( neighbor -> {
+                                    neighborsByEntitySet.get( associationEsId )
+                                            .add( neighbor.getAssociationEntityKeyId() );
+                                    neighborsByEntitySet.get( neighborEsId ).add( neighbor.getNeighborEntityKeyId() );
+                                }
+                        );
                     } );
                 } )
         );
 
-        List<AuditableEvent> events = new ArrayList<>( neighborsByEntitySet.keySet().size() + 1 );
-        UUID userId = spm.getCurrentUserId();
-
-        events.add( new AuditableEvent(
-                userId,
-                new AclKey( entitySetId ),
-                AuditEventType.LOAD_ENTITY_NEIGHBORS,
-                "Load neighbors of entities with filter through SearchApi.executeFilteredEntityNeighborSearch",
-                Optional.of( filter.getEntityKeyIds() ),
-                ImmutableMap.of( "filters", filter ),
-                OffsetDateTime.now(),
-                Optional.empty()
-        ) );
+        List<AuditableEvent> events = new ArrayList<>(
+                neighborsByEntitySet.keySet().size() + filter.getEntityKeyIds().size() );
+        UUID userId = getCurrentUserId();
```

Comment:
I think you accidentally removed the `spm.` in front of this

Reply:
Yep, thanks! :)